src/engine/Cache.class.js

```diff
@@ -421,36 +428,78 @@ class Cache {
+        if (!fs.existsSync(this.archiveFolder)) {
+          fs.mkdirSync(this.archiveFolder, { recursive: true })
+        }
+
+        // Move original file into the archive folder
+        fs.rename(filePath, archivePath, (renameError) => {
+          if (renameError) {
+            this.logger.error(renameError)
+          } else {
+            this.logger.info(`File ${filePath} moved to ${archivePath}`)
           }
+        })
+      } else {
+        // Delete original file
+        fs.unlink(filePath, (unlinkError) => {
+          if (unlinkError) {
+            this.logger.error(unlinkError)
+          } else {
+            this.logger.info(`File ${filePath} deleted`)
+          }
+        })
+      }
+    }
+  }
 
-          // Move original file into the archive folder
-          fs.rename(filePath, archivePath, (renameError) => {
-            if (renameError) {
-              this.logger.error(renameError)
-            } else {
-              this.logger.info(`File ${filePath} moved to ${archivePath}`)
-            }
-          })
-          break
-        default:
-          this.logger.error(`unknown Archive Mode: ${this.archiveMode}`)
+  /**
+   * Delete file in archiveFolder
+   * @param {string} filePath - The file
+   * @return {void}
+   */
+  async refreshArchiveFolder() {
+    this.logger.silly('refreshArchiveFolder()')
+    // if a process already occurs, it clears it
+    if (this.archiveTimeout) {
+      clearTimeout(this.archiveTimeout)
+    }
+
+    const files = fs.readdirSync(this.archiveFolder)
+    const timestamp = new Date().getTime()
+    if (files.length > 0) {
+      // Disable ESLint check because we need for..of loop to support async calls
+      // eslint-disable-next-line no-restricted-syntax
+      for (const file of files) {
+        // Disable ESLint check because we want to handle files one by one
+        // eslint-disable-next-line no-await-in-loop
+        const stats = fs.statSync(path.join(this.archiveFolder, file))
+
+        if (stats.mtimeMs + this.retentionDuration < timestamp) {
+          // local try catch in case an error occurs on a file
+          // if so, the loop goes on with the other files
+          try {
+            // eslint-disable-next-line no-await-in-loop
+            fs.unlink(path.join(this.archiveFolder, file), (unlinkError) => {
+              if (unlinkError) {
+                this.logger.error(unlinkError)
+              } else {
+                this.logger.info(`File ${path.join(this.archiveFolder, file)} deleted`)
+              }
+            })
+          } catch (sendFileError) {
+            this.logger.error(`Error sending the file ${file}: ${sendFileError.message}`)
+          }
+        }
       }
+    } else {
+      this.logger.debug(`The archive folder ${this.archiveFolder} is empty. Nothing to delete`)
     }
+    this.archiveTimeout = setTimeout(() => {
+      this.refreshArchiveFolder()
+    }, 3600000)
```

Comment:
Create a global const variable at the top of the file and use it here. It's easier if we need to change the duration later or just if we want to check its value without looking for the timeout